### PR TITLE
feat: implementing asset player with media-viewer

### PR DESCRIPTION
**Pull request by nuanyang233**
closes MASK-450

### CODE REVIEW

**Comment by wenluomask** [comment]
Refer to https://mask.atlassian.net/browse/MASK-450

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

File: packages/mask/src/plugins/Collectible/SNSAdaptor/ArticleTab.tsx [hunk 1]
```diff
@@ -15,44 +14,40 @@ const useStyles = makeStyles()({
...
+                            url={resourceUrl}
+                            options={{
+                                playsInline: true,
+                            }}
+                            classes={{
+                                errorPlaceholder: classes.errorPlaceholder,
+                                loadingPlaceholder: classes.loadingPlaceholder,
+                            }}
+                        />
+                    ) : null}
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Collectible/SNSAdaptor/ArticleTab.tsx:48]: Need a fallback UI here.

File: packages/mask/src/plugins/Collectible/SNSAdaptor/ArticleTab.tsx [hunk 2]
```diff
@@ -15,44 +14,40 @@ const useStyles = makeStyles()({
...
-                    <Link href={asset.value.animation_url} target="_blank" rel="noopener noreferrer">
-                        <AssetPlayer src={resourceUrl} alt={asset.value?.name ?? ''} />
-                    </Link>
-                ) : (
-                    <AssetPlayer src={resourceUrl} alt={asset.value?.name ?? ''} />
-                )}
-            </div>
-        </CollectibleTab>
-    )
+    return useMemo(() => {
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Collectible/SNSAdaptor/ArticleTab.tsx:31]: Why memorize a UI part?

File: packages/shared/src/UI/components/AssetPlayer/index.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                url,
+                type,
+                ...options,
+            })
+        }
+    }, [url, type, options, playerState])
+    //endregion
+
+    //#region resource loaded error
+    const onMessage = useCallback(({ message }: { message: any }) => {
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/AssetPlayer/index.tsx:85]: Avoid any.

**Comment by wenluomask** [comment]
Checked.

**Review by Lanttcat — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**